File: src/main/java/zd/zero/waifu/motivator/plugin/alert/notification/AlertConfiguration.java
package zd.zero.waifu.motivator.plugin.alert.notification;

import lombok.Builder;
import lombok.Getter;

@Getter
@Builder
public class AlertConfiguration {

    private final boolean isAlertEnabled;

    private final boolean isDisplayNotificationEnabled;

    private final boolean isSoundAlertEnabled;

}

File: src/main/java/zd/zero/waifu/motivator/plugin/alert/AlertConfiguration.kt
package zd.zero.waifu.motivator.plugin.alert

data class AlertConfiguration(
    val isAlertEnabled: Boolean = false,
    val isDisplayNotificationEnabled: Boolean = false,
    val isSoundAlertEnabled: Boolean = false
) {
    companion object {
        fun allEnabled(): AlertConfiguration =
            AlertConfiguration(
                isAlertEnabled = true,
                isDisplayNotificationEnabled = true,
                isSoundAlertEnabled = true
            )
    }
}
